tests/test_crystal_ball.py

```diff
@@ -86,103 +112,101 @@ def test_conversions():
     )
     assert examine.prepare(
         cpu=1, gpu=1, ram="20GB", disk="", jobs=1, job_duration="10m",
-        maxnodes=0, verbose=True, config_file=config_file
+        maxnodes=0, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=1, gpu=0, ram="10GB", disk="10GB", jobs=1, job_duration="10m",
-        maxnodes=0, verbose=True, config_file=config_file
+        maxnodes=0, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=1, gpu=0, ram="10GB", disk="10GB", jobs=128, job_duration="15m",
-        maxnodes=0, verbose=True, config_file=config_file
+        maxnodes=0, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=1, gpu=0, ram="10GB", disk="", jobs=1, job_duration="10m",
-        maxnodes=1, verbose=True, config_file=config_file
+        maxnodes=1, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=8, gpu=0, ram="10GB", disk="", jobs=1, job_duration="10m",
-        maxnodes=0, verbose=True, config_file=config_file
+        maxnodes=0, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=8, gpu=0, ram="80GB", disk="", jobs=4, job_duration="1h",
-        maxnodes=0, verbose=True, config_file=config_file
+        maxnodes=0, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=2, gpu=0, ram="10GB", disk="", jobs=1, job_duration="10m",
-        maxnodes=3, verbose=True, config_file=config_file
+        maxnodes=3, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=1, gpu=0, ram="20GB", disk="", jobs=1, job_duration="10m",
-        maxnodes=2, verbose=True, config_file=config_file
+        maxnodes=2, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=2, gpu=0, ram="20GB", disk="", jobs=1, job_duration="",
-        maxnodes=2, verbose=True, config_file=config_file
+        maxnodes=2, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=2, gpu=0, ram="20GB", disk="", jobs=32, job_duration="10m",
-        maxnodes=1, verbose=True, config_file=config_file
+        maxnodes=1, verbose=True, content=mocked_content
     )
     assert examine.prepare(
         cpu=2, gpu=5, ram="10GB", disk="", jobs=32, job_duration="10m",
-        maxnodes=1, verbose=True, config_file=config_file
+        maxnodes=1, verbose=True, content=mocked_content
     )
 
 
-def test_slot_config(root_dir):
+def test_slot_config():
     """
     Tests the slot loading method.
     :return:
     """
-    config_file = path.join(root_dir, 'example_config.json')
+    mocked_content = mocked_collector.query
 
-    with open(config_file) as f:
-        slots = json.load(f)['slots']
+    slots = collect.collect_slots(mocked_content)
 
-    assert "SlotType" in slots[0]["slot_size"][0]
+    assert "SlotType" in slots['cpu2']["slot_size"][0]
```

Comment:
This is the cause of the tests failing. `slots` is a list, but you assume it to be a dictionary. I think you _probably_ want to replace `"slot_size"` with `'cpu2'`, but check if this works with the slots config setup.